sdk/core/src/http_client/reqwest.rs

```diff
@@ -1,14 +1,24 @@
-use crate::error::{ErrorKind, ResultExt};
-use crate::{Body, HttpClient, PinnedStream};
-
+use crate::{
+    error::{ErrorKind, ResultExt},
+    Body, HttpClient, PinnedStream,
+};
 use async_trait::async_trait;
 use futures::TryStreamExt;
-use std::{collections::HashMap, str::FromStr};
+use std::{collections::HashMap, str::FromStr, sync::Arc};
 
 /// Construct a new `HttpClient` with the `reqwest` backend.
-pub fn new_reqwest_client() -> std::sync::Arc<dyn HttpClient> {
+pub fn new_reqwest_client() -> Arc<dyn HttpClient> {
     log::debug!("instantiating an http client using the reqwest backend");
-    std::sync::Arc::new(::reqwest::Client::new())
+
+    // set `pool_max_idle_per_host` to `0` to avoid an issue in the underlying
+    // `hyper` library that causes the `reqwest` client to hang in some cases.
+    //
+    // See <https://github.com/hyperium/hyper/issues/2312> for more details.
+    let client = ::reqwest::ClientBuilder::new()
+        .pool_max_idle_per_host(0)
```

Comment:
Are there any adverse side effects of setting this?

Reply:
It effectively disables connection pools.  This has a performance impact but mitigates a client hang.

In addition to the original reporter, I've experienced this issue as well.  It's been an open issue in the hyper repo since October 2020.